sling/task/dashboard.h

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SLING_TASK_DASHBOARD_H_
+#define SLING_TASK_DASHBOARD_H_
+
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "sling/base/types.h"
+#include "sling/file/file.h"
+#include "sling/http/http-server.h"
+#include "sling/http/static-content.h"
+#include "sling/task/job.h"
+#include "sling/util/mutex.h"
+
+namespace sling {
+namespace task {
+
+// Dashboard for monitoring jobs.
+class Dashboard : public Monitor {
+ public:
+  enum Status {IDLE, MONITORED, FINAL, SYNCHED, TERMINAL};
```

Comment:
I have added some comments to the different enum values to make this more clear. The status tracking is used synchronizing the dashboard client when all the jobs have completed. If the web client is polling the server (e.g. every 15 secs), it will see that the "finished" flag is set in the response when all jobs have completed and the final status messages will contain the final job status with peak values for some parameters (ram and temperature). The web client will then disable polling and display the final status and the workflow process can then terminate. This setup allows us to capture the final status in the browser without having to keep the workflow process running and the dashboard polling with automatically stop when the workflow is complete.